rllib/utils/spaces/space_utils.py

```diff
@@ -256,18 +256,18 @@ def unsquash_action(action, action_space_struct):
     """
 
     def map_(a, s):
-        if (
-            isinstance(s, gym.spaces.Box)
-            and (s.dtype == np.float32 or s.dtype == np.float64)
-            and np.all(s.bounded_below)
-            and np.all(s.bounded_above)
-        ):
-            # Assuming values are roughly between -1.0 and 1.0 ->
-            # unsquash them to the given bounds.
-            a = s.low + (a + 1.0) * (s.high - s.low) / 2.0
-            # Clip to given bounds, just in case the squashed values were
-            # outside [-1.0, 1.0].
-            a = np.clip(a, s.low, s.high)
+        if isinstance(s, gym.spaces.Box) and np.all(s.bounded_below) and np.all(s.bounded_above):
+            if s.dtype == np.float32 or s.dtype == np.float64:
+                # Assuming values are roughly between -1.0 and 1.0 ->
+                # unsquash them to the given bounds.
+                a = s.low + (a + 1.0) * (s.high - s.low) / 2.0
+                # Clip to given bounds, just in case the squashed values were
+                # outside [-1.0, 1.0].
+                a = np.clip(a, s.low, s.high)
+            elif s.dtype == np.int32 or s.dtype == np.int64:
```

Comment:
The comments make reading this super easy :)
Should we check for np.integer here instead?
np.issubdtype(s.dtype, np.integer)?

Reply:
done. it's weird, why doesn't
```
np.issubdtype(s.dtype, np.float)
```
work for float32 and float64??

Reply:
Huh. Works on my machine.

Reply:
s = np.array([0, 1], dtype=np.float64)
np.issubdtype(s.dtype, np.float)
True

Reply:
numpy==1.22.3

Reply:
> numpy==1.22.3

right, but do you know why float32 doesn't work?
```
>>> s = np.array([0, 1], dtype=np.float32)
>>> np.issubdtype(s.dtype, np.float)
False
```